components/raftstore/src/store/peer.rs

```diff
@@ -4058,6 +4069,9 @@ where
             // The message is dropped silently, this usually due to leader absence
             // or transferring leader. Both cases can be considered as NotLeader error.
             return Err(Error::NotLeader(self.region_id, None));
+        } else if self.force_leader {
+            // forward the commit index
+            self.raft_group.raft.raft_log.committed = self.raft_group.raft.raft_log.last_index();
```

Comment:
This breaks the assumption that only persisted entries get committed.

Reply:
change it